nullaway/src/main/java/com/uber/nullaway/qual/EnsuresNonNull.java

```diff
@@ -0,0 +1,19 @@
+package com.uber.nullaway.qual;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Can annotate a methods with @EnsuresNonnull(param) annotation where param is one of the classes
+ * fields. It indicates a post-condition for the method, that at every call site to this method, the
+ * class field in the argument is @Nonnull at exit point. If a method is annotated
+ * with @EnsuresNonnull(param), NullAway checks weather the @Nonnull assumption of the field at exit
+ * point is valid.
+ */
+@Retention(RetentionPolicy.CLASS)
+@Target({ElementType.METHOD, ElementType.CONSTRUCTOR})
+public @interface EnsuresNonNull {
```

Comment:
Why are these qualifiers getting added in this PR?

Reply:
This was strange to me as well, I am not sure what is going on. I did pull latest changes of upstream master branch and still this is part of this PR!

Reply:
Hi @msridhar for some reason Github does not allow me to reply to this https://github.com/uber/NullAway/pull/517#discussion_r777670191, so I will answer it here.

Below is the activation status of these two flags:

* ```logErrorEnabled ```: ```AutoFixer``` always activates it in all runs of ```NullAway```.
* ```suggestEnabled```: ```AutoFixer``` runs ```NullAway``` with this flag activated only once per iteration and deactivates in ```Diagnosing``` phase. In ```Deep Search``` mode, ```AutoFixer``` keeps it activated in ```Diagnosing``` phase as well.



I can remove the flag and always log errors when ```WriteFixMetadata``` is active. 

It may seem the creation of ```FixSerializationConfig``` is unnecessary, however, for finalizing ```AutoFixer``` we need to create the two files below which is not part of this PR for now but will be required in next PR.

1. We need to write metadata regarding ```@Nonnull``` fields at exit point of methods to detect ```@Initializers```.
2. Examine protection of methods against nullability of parameters.

These different modes to run ```NullAway``` are necessary for ```AutoFixer``` which will be added in the upcoming PR  and are controlled in and this is my motivation for having extra config file. Please let me know your thoughts on this and let me know if you would like to remove ```FixSerializationConfig``` file.

Reply:
I believe this is ready for another round of review. @msridhar

Reply:
> Why are these qualifiers getting added in this PR?

I am sorry I figured this out, it was duplicates. The directory was renamed to ```annotations```  on master branch and in my branch it was in ```qual``` directory.